agent/src/beerocks/monitor/monitor_stats.cpp

```diff
@@ -457,6 +460,58 @@ bool monitor_stats::add_ap_metrics(ieee1905_1::CmduMessageTx &cmdu_tx, const sMa
     return true;
 }
 
+bool monitor_stats::add_ap_assoc_sta_traffic_stat(ieee1905_1::CmduMessageTx &cmdu_tx,
+                                                  monitor_sta_node &sta_node)
+{
+    auto ap_assoc_sta_traffic_stat_tlv = cmdu_tx.addClass<wfa_map::tlvAssociatedStaTrafficStats>();
+    if (!ap_assoc_sta_traffic_stat_tlv) {
+        LOG(ERROR) << "Couldn't addClass tlvAssociatedStaTrafficStats";
+        return false;
+    }
+    auto stat                                        = sta_node.get_stats().hal_stats;
+    ap_assoc_sta_traffic_stat_tlv->sta_mac()         = tlvf::mac_from_string(sta_node.get_mac());
+    ap_assoc_sta_traffic_stat_tlv->byte_sent()       = stat.tx_bytes;
+    ap_assoc_sta_traffic_stat_tlv->byte_recived()    = stat.rx_bytes;
+    ap_assoc_sta_traffic_stat_tlv->packets_sent()    = stat.tx_packets;
+    ap_assoc_sta_traffic_stat_tlv->packets_recived() = stat.rx_packets;
+    //TODO: add tx_packets_error in bwl::SStaStats
+    ap_assoc_sta_traffic_stat_tlv->tx_packets_error() = 0;
+    //TODO: rx_packets_error in bwl::SStaStats
+    ap_assoc_sta_traffic_stat_tlv->rx_packets_error()     = 0;
+    ap_assoc_sta_traffic_stat_tlv->retransmission_count() = stat.retrans_count;
+
+    return true;
+}
+
+bool monitor_stats::add_ap_assoc_sta_link_metric(ieee1905_1::CmduMessageTx &cmdu_tx,
+                                                 const sMacAddr &bssid, monitor_sta_node &sta_node)
+{
+    auto ap_assoc_sta_link_metric_tlv = cmdu_tx.addClass<wfa_map::tlvAssociatedStaLinkMetrics>();
+    if (!ap_assoc_sta_link_metric_tlv) {
+        LOG(ERROR) << "Couldn't addClass tlvAssociatedStaLinkMetrics";
+        return false;
+    }
+
+    ap_assoc_sta_link_metric_tlv->sta_mac() = tlvf::mac_from_string(sta_node.get_mac());
+
+    // there is exactly 1 BSSID per STA
```

Comment:
I can't understand how this comment supports the code below it.
maybe:
```suggestion
    // default allocation creates a list with one element which it what we want: exactly 1 BSSID per STA
```